lib/alex_inject.rb

```diff
@@ -0,0 +1,26 @@
+class Array
+
+  def alex_inject(result = nil)
+    duplicate = self.dup
+    if result == nil; result = duplicate[0]; duplicate.delete_at(0); end
```

Comment:
Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`.<br>Style/IfWithSemicolon: Do not use if x; Use the ternary operator instead.<br>Style/NilComparison: Prefer the use of the `nil?` predicate.